dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/plugins/BaseTokenHandler.java

```diff
@@ -158,6 +166,29 @@ public OAuth20Token createToken(Map<String, String[]> tokenMap) {
         return token;
     }
 
+    private Map<String, Object> getThirdPartyClaims(String thirdPartyAccessToken, OidcServerConfig oidcServerConfig) {
+        Map<String, Object> thirdPartyAccessTokenClaims = new HashMap<String, Object>();
+        if (thirdPartyAccessToken == null || thirdPartyAccessToken.isEmpty()) {
+            return thirdPartyAccessTokenClaims;
+        }
+        try {
+            Set<String> allowedThirdPartyAccessTokenClaims = oidcServerConfig.getThirdPartyAccessTokenClaims();
+
+            JwtContext jwtContext = Jose4jUtil.parseJwtWithoutValidation(thirdPartyAccessToken);
+            JwtClaims jwtClaims = jwtContext.getJwtClaims();
+            Map<String, Object> jwtClaimsMap = jwtClaims.getClaimsMap();
+
+            for (String thirdPartyClaim : allowedThirdPartyAccessTokenClaims) {
+                if (jwtClaimsMap.containsKey(thirdPartyClaim)) {
+                    thirdPartyAccessTokenClaims.put(thirdPartyClaim, jwtClaimsMap.get(thirdPartyClaim));
+                }
+            }
+        } catch (Exception e) {
+            // ignore for now
```

Comment:
We'll probably want to consider adding an NLS warning message in here.

Reply:
an nls warning will be added in a follow-up pr